src/components/password/password.component.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useRef, useState } from "react";
+import { TextboxProps } from "../textbox";
+import { StyledPassword, HiddenAriaLive } from "./password.style";
+import guid from "../../__internal__/utils/helpers/guid";
+import useLocale from "../../hooks/__internal__/useLocale";
+import ButtonMinor from "../button-minor/button-minor.component";
+
+export type TypeOptions = "text" | "password";
+
+export interface PasswordProps extends TextboxProps {
+  /** HTML type attribute of the input */
+  type?: TypeOptions;
+}
+
+export const Password = ({ id, type, ...rest }: PasswordProps) => {
+  const internalInputId = useRef(id || guid());
+  const l = useLocale();
+
+  const [passwordShown, setPasswordShown] = useState(false);
+
+  return (
+    <>
+      <StyledPassword
+        data-element="styled-password-container"
+        id={internalInputId.current}
+        autoComplete="off"
+        type={passwordShown ? "text" : "password"}
+        {...rest}
+      >
+        <ButtonMinor
```

Comment:
Might be an idea to destructure `disabled` prop and pass it ButtonMinor as when the component is disabled, the button still works.